### ISSUE TITLE
Segfault launching a football game

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#2769]
When I launch a football game, it sometimes segfault.

From my tests, it is related to the number of AI karts. Works fine with 1, 2, 4 6, 7 ; segfault with 3, 5 ; with a 100% correlation. [Edit : After more tests while trying to produce the debugging output, it appears more random than first reported. Sometimes happen, sometimes not, but a few tries always trigger it.]

**Comment by Alayan-stk-2** [comment]
gdb output : 

0  0x00007ff3f327a428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
1  0x00007ff3f327c02a in __GI_abort () at abort.c:89
2  0x00007ff3f3272bd7 in __assert_fail_base (fmt=<optimized out>, assertion=assertion@entry=0x14df950 "m_blue_kdm.size() > 0 && m_red_kdm.size() > 0", 
    file=file@entry=0x14df8d8 "/home/*/stk-code-Parachute/src/modes/soccer_world.hpp", line=line@entry=384, 
    function=function@entry=0x14e05a0 <SoccerWorld::getBallChaser(SoccerTeam) const::__PRETTY_FUNCTION__> "int SoccerWorld::getBallChaser(SoccerTeam) const") at assert.c:92
3  0x00007ff3f3272c82 in __GI___assert_fail (assertion=0x14df950 "m_blue_kdm.size() > 0 && m_red_kdm.size() > 0", file=0x14df8d8 "/home/*/stk-code-Parachute/src/modes/soccer_world.hpp", line=384, 
    function=0x14e05a0 <SoccerWorld::getBallChaser(SoccerTeam) const::__PRETTY_FUNCTION__> "int SoccerWorld::getBallChaser(SoccerTeam) const") at assert.c:101
4  0x0000000000d6a706 in SoccerWorld::getBallChaser (this=0x21ddd460, team=SOCCER_TEAM_RED) at /home/*/stk-code-Parachute/src/modes/soccer_world.hpp:384
5  0x0000000000d68481 in SoccerAI::findTarget (this=0x22cdab80) at /home/*/stk-code-Parachute/src/karts/controller/soccer_ai.cpp:186
6  0x0000000000d634ec in ArenaAI::update (this=0x22cdab80, dt=0.0500000007) at /home/*/stk-code-Parachute/src/karts/controller/arena_ai.cpp:125
7  0x0000000000d681f5 in SoccerAI::update (this=0x22cdab80, dt=0.0500000007) at /home/*/stk-code-Parachute/src/karts/controller/soccer_ai.cpp:132
8  0x0000000000d3be49 in Kart::update (this=0x28a156a0, dt=0.0500000007) at /home/*/stk-code-Parachute/src/karts/kart.cpp:1238
9  0x0000000000e15410 in World::update (this=0x21ddd460, dt=0.0500000007) at /home/*/stk-code-Parachute/src/modes/world.cpp:989
10 0x0000000000e086e6 in SoccerWorld::update (this=0x21ddd460, dt=0.0500000007) at /home/*/stk-code-Parachute/src/modes/soccer_world.cpp:192
11 0x0000000000e14d3c in World::updateWorld (this=0x21ddd460, dt=0.0500000007) at /home/*/stk-code-Parachute/src/modes/world.cpp:861
12 0x0000000000bb6e2c in MainLoop::updateRace (this=0x3534000, dt=0.0500000007) at /home/*/stk-code-Parachute/src/main_loop.cpp:151
13 0x0000000000bb6fd5 in MainLoop::run (this=0x3534000) at /home/*/stk-code-Parachute/src/main_loop.cpp:246
14 0x0000000000af065e in main (argc=1, argv=0x7ffe8c7d5338) at /home/*/stk-code-Parachute/src/main.cpp:1771

**Comment by auriamg** [comment]
I have been unable to reproduce the issue so far, any more information on how to reproduce?

**Comment by Alayan-stk-2** [comment]
@Benau too has been unable to reproduce the issue.

I did some debugging with him yesterday and in addition to the aboe gdb log, I found that the values evaluated, which should be >0, are equal to 0 when it triggers the crash, but I don't know more.

After some more testing, I got it to crash with other numbers of AI, so I don't think the problem stems from there.

It may crash at the first try (launching the game, and following the more direct steps to the football game and launching it)
